### PR TITLE
Create a helper hdr_setup function for CBMC proofs

### PR DESCRIPTION
**Pull request by tegansb**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: verification/cbmc/include/make_common_data_structures.h [hunk 1]
```diff
@@ -46,6 +46,10 @@ struct aws_cryptosdk_hdr *ensure_nondet_hdr_has_allocated_members(const size_t m
 bool aws_cryptosdk_hdr_members_are_bounded(
     const struct aws_cryptosdk_hdr *hdr, const size_t max_edk_item_size, const size_t max_item_size);
 
+/* Allocates and ensures properties of the members of the header. */
+struct aws_cryptosdk_hdr *hdr_setup(
+    const size_t max_table_size, const size_t max_edk_item_size, const size_t max_item_size);
+
```
Comment (feliperodri) [inline verification/cbmc/include/make_common_data_structures.h:52]: In the header documentation, I'd make absolutely clear all assumptions this function encapsulates.

**Review by feliperodri — APPROVED** [review]
LGTM, but I'd update documentation to make clear about all assumptions in this function. We want to make them as visible as possible.

File: verification/cbmc/sources/make_common_data_structures.c [hunk 2]
```diff
@@ -177,6 +177,20 @@ bool aws_cryptosdk_hdr_members_are_bounded(
     return true; /* If hdr is NULL, true by default */
 }
 
+struct aws_cryptosdk_hdr *hdr_setup(
+    const size_t max_table_size, const size_t max_edk_item_size, const size_t max_item_size) {
```
Comment (adpaco-aws) [inline verification/cbmc/sources/make_common_data_structures.c:181]: `max_item_size` is not used anywhere.

**Review by adpaco-aws — CHANGES_REQUESTED** [review]

**Review by alex-chew — APPROVED** [review]
I agree with Felipe's and Adrian's previous comments, otherwise LGTM.

File: verification/cbmc/sources/make_common_data_structures.c (hunk 2 above)
Comment (tegansb) [reply verification/cbmc/sources/make_common_data_structures.c:181]: Thanks! copy-paste error which is fixed now!

**Review by adpaco-aws — APPROVED** [review]

File: verification/cbmc/include/make_common_data_structures.h (hunk 1 above)
Comment (tegansb) [reply verification/cbmc/include/make_common_data_structures.h]: Good point! I've added documentation that explicitly lists each assumption make by hdr_setup.

**Pull request merged**